src/interfaces/erc165.cairo

```diff
@@ -1,50 +1,4 @@
-const IERC165_ID: u32 = 0x01ffc9a7_u32;
-const INVALID_ID: u32 = 0xffffffff_u32;
-
 #[starknet::interface]
-trait IERC165<TContractState> {
-    fn supportsInterface(self: @TContractState, interface_id: u32) -> bool;
-}
-
-#[starknet::contract]
-mod ERC165 {
-    //
-    // Storage
-    //
-
-    #[storage]
-    struct Storage {
-        supported_interfaces: LegacyMap<u32, bool>
-    }
-
-    //
-    // ERC165 impl
-    //
-
-    #[external(v0)]
-    impl ERC165Impl of super::IERC165<ContractState> {
-        fn supportsInterface(self: @ContractState, interface_id: u32) -> bool {
-            if interface_id == super::IERC165_ID {
-                return true;
-            }
-            self.supported_interfaces.read(interface_id)
-        }
-    }
-
-    //
-    // Internal
-    //
-
-    #[generate_trait]
-    impl HelperImpl of HelperTrait {
-        fn register_interface(ref self: ContractState, interface_id: u32) {
-            assert(interface_id != super::INVALID_ID, 'Invalid id');
-            self.supported_interfaces.write(interface_id, true);
-        }
-
-        fn deregister_interface(ref self: ContractState, interface_id: u32) {
-            assert(interface_id != super::IERC165_ID, 'Invalid id');
-            self.supported_interfaces.write(interface_id, false);
-        }
-    }
+trait IERC165<T> {
```

Comment:
Why `TContractState` -> `T`? `TContractState` has the advantage to provide some guidance for dev.

Reply:
`T` might change in the future, especially in this particular case where with components
cf [this code](https://github.com/starkware-libs/cairo/blob/b948fd96f5a16a29d48001ca33f21fd499a87fbc/crates/cairo-lang-starknet/test_data/test_contract.cairo#L4)